### PR TITLE
Provide different tests for JSON output 

### PR DESCRIPTION
**Pull request by linev**
Included are several classes; testing most kinds of streamer elements.
Will work only after merging https://github.com/root-project/root/pull/580

### CODE REVIEW

File: root/io/json/Makefile [hunk 1]
```diff
@@ -66,9 +66,26 @@ testWithDiff: testWithDiff.log testWithDiff.ref
...
+	$(TestDiff)
+
+STL: STL.log
+	$(TestDiff)
+
+StreamerLoop: StreamerLoop.log
+	$(TestDiff)
+
+RootClasses: RootClasses.log
+	$(TestDiff)
```
Comment (linev) [inline root/io/json/Makefile]: I add correspondent rule in Makefile

File: root/io/json/test_classes.h [hunk 2]
```diff
@@ -0,0 +1,852 @@
...
+           fInt[n] = n*5;
+           fLong[n] = n*123;
+           fFloat[n] = n*3;
+           fDouble[n] = n*7;
+        }
+
+     }
+     virtual ~TJsonEx3()
+     {
+       //delete[] fBool;
```
Comment (linev) [inline root/io/json/test_classes.h]: This was due to missing copy constructor. 
I used this class in std::vector and arrays pointers were copied 
Now class should work correctly.

File: root/io/json/test_classes.h [hunk 3]
```diff
@@ -0,0 +1,891 @@
...
+     std::string  fStringArr234[2][3][4];
+
+     TJsonEx2*   GetObj2() { return fObj2; }
+     TJsonEx2**  GetPtr2() { return fPtr2; }
+     TJsonEx2**  GetSafePtr2() { return fSafePtr2; }
+
+     TJsonEx6(bool setvalues = false)
+     {
+        for (int n=0;n<3;n++) {
+           fPtr1[n] = 0;
```
Comment (pcanal) [inline root/io/json/test_classes.h]: whereever you have 0 meaning null pointer, can you use 'nullptr'?  thanks.

Comment (linev) [inline root/io/json/test_classes.h]: Done

**Comment by linev** [comment]
Philippe - you are too fast, 
Test will fail with master branch.
You first should merge https://github.com/root-project/root/pull/580

**Pull request merged**